server/hedley/modules/custom/hedley_reports/scripts/generate-completion-data.php

```diff
@@ -0,0 +1,93 @@
+<?php
+
+/**
+ * @file
+ * Generates completion data for different types of encounters.
+ *
+ * Execution: drush scr
+ *   profiles/hedley/modules/custom/hedley_reports/scripts/generate-completion-data.php.
+ */
+
+if (!drupal_is_cli()) {
+  // Prevent execution from browser.
+  return;
+}
+
+// Get the last node id.
+$nid = drush_get_option('nid', 0);
+
+// Get the number of nodes to be processed.
+$batch = drush_get_option('batch', 50);
+
+// Flag to generate data only if it was not generated already.
+$exclude_set = drush_get_option('exclude_set', FALSE);
+
+// Get allowed memory limit.
+$memory_limit = drush_get_option('memory_limit', 500);
+
+$type = 'nutrition_encounter';
+$base_query = new EntityFieldQuery();
+$base_query
+  ->entityCondition('entity_type', 'node')
+  ->entityCondition('bundle', $type)
+  ->propertyCondition('status', NODE_PUBLISHED);
+
+if ($exclude_set) {
+  $base_query->addTag('exclude_set_reports_data');
+}
+
+$count_query = clone $base_query;
+$count_query->propertyCondition('nid', $nid, '>');
+$count = $count_query->count()->execute();
+
+if ($count == 0) {
+  drush_print("There are no nodes of type $type in DB.");
+  exit;
+}
+
+// @todo: do we need this?
+$nurses = hedley_ncda_resolve_nurses_ids();
+
+$total = 0;
+drush_print("$count nodes of type $type located.");
+
+while (TRUE) {
+  $query = clone $base_query;
+  if ($nid) {
+    $query->propertyCondition('nid', $nid, '>');
+  }
+
+  $result = $query
+    ->range(0, $batch)
+    ->execute();
+
+  if (empty($result['node'])) {
+    // No more items left.
+    break;
+  }
+
+  $ids = array_keys($result['node']);
+  $nodes = node_load_multiple($ids);
+  foreach ($nodes as $node) {
+    $completion_data = hedley_reports_generate_completion_data_for_nutrition_encounter($node);
+    $node->field_reports_data[LANGUAGE_NONE][0]['value'] = json_encode($completion_data);
+    node_save($node);
```

Comment:
should we wrap in a try-catch and add error handling upon failure? Or ok to let it fail as-is?

Reply:
@AronNovak 

At https://github.com/TIP-Global-Health/eheza-app/pull/1283/files#diff-576b4cc9568e1d723be865511b32f6fcb9147fdf926905d82686213e1435548dR23.

The context is that every encounter has to calculate it's completion data only once.
From that point on, it never changes.

So, the plan is to run the script once initially, and then, have overnight Jenkins task that will fill completion data for only those encounters, that don't have it set.

Reply:
Don't see a real need for this. `node_save()` failures are very rare.
If it will happen, the calculation will be attempted again the next night.
As we deal with thousands of encounters, final completion results will not be affected.